packages/datetime/src/components/date-range-input/dateRangeInput.tsx

```diff
@@ -630,10 +630,23 @@ export class DateRangeInput extends AbstractPureComponent<DateRangeInputProps, D
     private handleInputKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
         const isTabPressed = e.key === "Tab";
         const isEnterPressed = e.key === "Enter";
+        const isEscapeKeyPressed = e.key === "Escape";
         const isShiftPressed = e.shiftKey;
 
         const { selectedStart, selectedEnd } = this.state;
 
+        if (isEscapeKeyPressed) {
+            const { isStartInputFocused, isEndInputFocused } = this.state;
+            if (isStartInputFocused) {
+                this.startInputElement?.blur();
+            }
+            if (isEndInputFocused) {
+                this.endInputElement?.blur();
+            }
```

Comment:
Any reason we can't just optimistically blur both components without doing the state check? I think this would simplify things and also make sure that in case of weird bugs where the state gets out of sync we still handle this case properly.

Reply:
Fair point! Added in b4490dcaf83017a49f2e4b23dcb386e0856a4c7f